Centaurus.Domain/Global.cs

```diff
@@ -211,5 +237,71 @@ private static async Task ApplyUpdates()
         private static Timer snapshotTimoutTimer;
 
         private static Timer snapshotRunTimer;
+
+        #region Analytics
+
+        private static async void AnalyticsManager_OnUpdate()
+        {
+            var updates = new Dictionary<BaseSubscription, SubscriptionUpdateBase>();
+            foreach (var subs in InfoConnectionManager.GetActiveSubscriptions())
+            {
+                switch (subs)
+                {
+                    case DepthsSubscription depthsSubscription:
+                        {
+                            var depth = AnalyticsManager.MarketDepthsManager.GetDepth(depthsSubscription.Market, depthsSubscription.Precision);
+
+                            if (depth != null)
+                                updates.Add(depthsSubscription, new MarketDepthUpdate { MarketDepth = depth, UpdateDate = depth.UpdatedAt });
+                        }
+                        break;
+                    case PriceHistorySubscription priceHistorySubscription:
+                        {
+                            var frames = await AnalyticsManager.OHLCManager.GetFrames(0, priceHistorySubscription.Market, priceHistorySubscription.FramePeriod);
```

Comment:
Where do you hold the cursor for current subscription? Looks like you grab the entire history every time.